src/main/java/org/sonar/plugins/stash/issue/MarkdownPrinter.java

```diff
@@ -31,30 +43,38 @@ public static String printSeverityMarkdown(String severity) {
     return sb.toString();
   }
 
-  public static String printIssueNumberBySeverityMarkdown(SonarQubeIssuesReport report, CoverageIssuesReport coverageReport, String severity) {
+  public static String printIssueNumberBySeverityMarkdown(List<Issue> report, String severity) {
     StringBuilder sb = new StringBuilder();
-    int issueNumber = report.countIssues(severity) + coverageReport.countLoweredIssues(severity);
+    long issueNumber = countIssuesBySeverity(report, severity);
     sb.append("| ").append(severity).append(" | ").append(issueNumber).append(" |").append(NEW_LINE);
 
     return sb.toString();
   }
 
-  public static String printIssueListBySeverityMarkdown(SonarQubeIssuesReport report, String sonarQubeURL, String severity) {
+  public static String printIssueListBySeverityMarkdown(List<Issue> report, String sonarQubeURL, String severity) {
     StringBuilder sb = new StringBuilder();
 
-    Map<String, SonarQubeIssue> rules = report.getUniqueRulesBySeverity(severity);
+    Map<String, Issue> rules = getUniqueRulesBySeverity(report, severity);
 
     // applying squid:S2864 optimization
-    for (Map.Entry<String, SonarQubeIssue> rule : rules.entrySet()) {
-      SonarQubeIssue issue = rule.getValue();
-      sb.append("| ").append(issue.printIssueMarkdown(sonarQubeURL)).append(" |").append(NEW_LINE);
+    for (Map.Entry<String, Issue> rule : rules.entrySet()) {
+      Issue issue = rule.getValue();
+      sb.append("| ").append(printIssueMarkdown(issue, sonarQubeURL)).append(" |").append(NEW_LINE);
     }
 
     return sb.toString();
   }
-  
-  public static String printReportMarkdown(PullRequestRef pr, String stashURL, String sonarQubeURL, SonarQubeIssuesReport report,
-                                           CoverageIssuesReport coverageReport, int issueThreshold) {
+
+  public static String printIssueMarkdown(Issue issue, String sonarQubeURL) {
+    StringBuilder sb = new StringBuilder();
+    sb.append(MarkdownPrinter.printSeverityMarkdown(issue.severity())).append(issue.message()).append(" [[").append(issue.ruleKey())
+            .append("]").append("(").append(sonarQubeURL).append("/").append(MarkdownPrinter.CODING_RULES_RULE_KEY).append(issue.ruleKey()).append(")]");
+
+    return sb.toString();
+  }
+
+  public static String printReportMarkdown(PullRequestRef pr, String stashURL, String sonarQubeURL, List<Issue> allIssues,

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Method has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00107)